src/backchannel_logout.rs

```diff
@@ -0,0 +1,495 @@
+//! This module implements components needed for [back-channel logout]
+//!
+//! [back-channel logout]: <https://openid.net/specs/openid-connect-backchannel-1_0.html>
+
+use std::collections::HashMap;
+
+use chrono::{DateTime, Utc};
+use serde::{de::Error, Deserialize};
+
+use crate::{
+    helpers::{FilteredFlatten, FlattenFilter},
+    types::helpers::{deserialize_string_or_vec, serde_utc_seconds},
+    types::SessionIdentifier,
+    AdditionalClaims, Audience, IssuerUrl, JsonWebKeyId, SubjectIdentifier,
+};
+
+/// The Logout Token as defined in [section 2.4] of the [OpenID Connect Back-Channel Logout spec][1]
+///
+/// [section 2.4]: <https://openid.net/specs/openid-connect-backchannel-1_0.html#LogoutToken>
+/// [1]: <https://openid.net/specs/openid-connect-backchannel-1_0.html>
+#[derive(Debug, Clone, PartialEq)]
+pub struct LogoutTokenClaims<AC: AdditionalClaims> {
+    /// The issuer of this token
+    iss: IssuerUrl,
+    /// The audience this token is intended for
+    aud: Vec<Audience>,
+    /// Time at which this token was issued
+    iat: DateTime<Utc>,
+    /// The unique identifier for this token. This can be used to detect
+    /// replay attacks.
+    jti: JsonWebKeyId,
+    identifier: LogoutIdentifier,
+    events: HashMap<String, serde_json::Value>,
+    additional_claims: FilteredFlatten<Self, AC>,
+}
+
+impl<AC> FlattenFilter for LogoutTokenClaims<AC>
+where
+    AC: AdditionalClaims,
+{
+    fn should_include(field_name: &str) -> bool {
+        !matches!(
+            field_name,
+            "iss" | "aud" | "iat" | "jti" | "sub" | "sid" | "events"
+        )
+    }
+}
```

Comment:
I don't think we need support for localization or something like that, right?

Reply:
I don't see any mentions of that in the spec, so hopefully not :-)